pkg/util/ranger/types_test.go

```diff
@@ -238,3 +252,366 @@ func TestRangeMemUsage(t *testing.T) {
+	range2.HighExclude = true
+	intersection1, _ = range1.IntersectRange(ctx.GetRangerCtx().TypeCtx, &range2)
+	actualResult[range1.String()+" "+range2.String()] = rangeToString(intersection1)
+	intersection2, _ = range2.IntersectRange(ctx.GetRangerCtx().TypeCtx, &range1)
+	require.Equal(t, intersection1, intersection2)
+	// Use of inifinity in ragnes. Intersection of [-infinity, 5] and [2, 14]
+	range1 = ranger.Range{
+		LowVal:    []types.Datum{types.NewIntDatum(math.MinInt64)},
+		HighVal:   []types.Datum{types.NewIntDatum(5)},
+		Collators: collate.GetBinaryCollatorSlice(1),
+	}
+	range2 = ranger.Range{
+		LowVal:    []types.Datum{types.NewIntDatum(2)},
+		HighVal:   []types.Datum{types.NewIntDatum(14)},
+		Collators: collate.GetBinaryCollatorSlice(1),
+	}
+	intersection1, _ = range1.IntersectRange(ctx.GetRangerCtx().TypeCtx, &range2)
+	actualResult[range1.String()+" "+range2.String()] = rangeToString(intersection1)
+	intersection2, _ = range2.IntersectRange(ctx.GetRangerCtx().TypeCtx, &range1)
+	require.Equal(t, intersection1, intersection2)
+
+	// Multi column range. Intersection of [1 2,1 5] (1 3,1 4] = (1 3,1 4]
+	range1 = ranger.Range{
+		LowVal:    []types.Datum{types.NewIntDatum(1), types.NewIntDatum(2)},
+		HighVal:   []types.Datum{types.NewIntDatum(1), types.NewIntDatum(5)},
+		Collators: collate.GetBinaryCollatorSlice(2),
+	}
+	range2 = ranger.Range{
+		LowVal:    []types.Datum{types.NewIntDatum(1), types.NewIntDatum(3)},
+		HighVal:   []types.Datum{types.NewIntDatum(1), types.NewIntDatum(4)},
+		Collators: collate.GetBinaryCollatorSlice(2),
+	}
+	range2.LowExclude = true
+	intersection1, _ = range1.IntersectRange(ctx.GetRangerCtx().TypeCtx, &range2)
+	actualResult[range1.String()+" "+range2.String()] = rangeToString(intersection1)
+	intersection2, _ = range2.IntersectRange(ctx.GetRangerCtx().TypeCtx, &range1)
+	require.Equal(t, intersection1, intersection2)
+
+	// Multi column range with different number of columns. Intersection of [1 1 -inf,1 1 15] [1 1 4,1 1 25] is [1 1 4,1 1 15]
+	range1 = ranger.Range{
+		LowVal:    []types.Datum{types.NewIntDatum(1), types.NewIntDatum(1), types.NewIntDatum(math.MinInt64)},
+		HighVal:   []types.Datum{types.NewIntDatum(1), types.NewIntDatum(1), types.NewIntDatum(15)},
+		Collators: collate.GetBinaryCollatorSlice(3),
+	}
+	range2 = ranger.Range{
+		LowVal:    []types.Datum{types.NewIntDatum(1), types.NewIntDatum(1), types.NewIntDatum(4)},
+		HighVal:   []types.Datum{types.NewIntDatum(1), types.NewIntDatum(1), types.NewIntDatum(25)},
+		Collators: collate.GetBinaryCollatorSlice(3),
+	}
+	intersection1, _ = range1.IntersectRange(ctx.GetRangerCtx().TypeCtx, &range2)
+	actualResult[range1.String()+" "+range2.String()] = rangeToString(intersection1)
+	intersection2, _ = range2.IntersectRange(ctx.GetRangerCtx().TypeCtx, &range1)
+	require.Equal(t, intersection1, intersection2)
+
+	// Compare actual vs expected
+	for k, v := range actualResult {
+		require.Equal(t, v, expectedResult[k])
```

Comment:
The expected should be the first argument.
```suggestion
		require.Equal(t, expectedResult[k], v)
```

Reply:
OK. done